manta-util/src/collections/vec_deque.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A double-ended queue (deque) implemented with a growable ring buffer.
+
+use crate::{array::BoxArray, collections::VecDeque};
+
+#[cfg(all(feature = "serde-alloc", feature = "serde-array"))]
+use crate::serde::{Deserialize, Serialize};
+
+#[doc(inline)]
+pub use alloc::collections::vec_deque::*;
+
+/// Multi-[`VecDeque`]
+#[cfg_attr(
+    all(feature = "serde-alloc", feature = "serde-array"),
+    derive(Deserialize, Serialize),
+    serde(
+        bound(deserialize = "T: Deserialize<'de>", serialize = "T: Serialize"),
+        crate = "crate::serde",
+        deny_unknown_fields
+    )
+)]
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct MultiVecDeque<T, const N: usize>(BoxArray<VecDeque<T>, N>);
+
+impl<T, const N: usize> MultiVecDeque<T, N> {
+    /// Builds a new empty [`MultiVecDeque`].
+    #[inline]
+    pub fn new() -> Self {
+        Self::default()
+    }
```

Comment:
Why doesn't new take any inputs?

Reply:
Because there is nothing to assign when creating a new container. It's like `Vec::new`.